File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/tasklistview/TaskGridItemViewAdapter.java
package com.wiseassblog.samsaradayplanner.ui.tasklistview;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.BaseAdapter;

import androidx.annotation.NonNull;

import com.wiseassblog.samsaradayplanner.common.ColorUtility;
import com.wiseassblog.samsaradayplanner.common.IconUtility;
import com.wiseassblog.samsaradayplanner.domain.Task;

class TaskGridItemViewAdapter extends BaseAdapter implements AdapterView.OnItemClickListener {

    private final Task[] taskItems;
    private TaskGridAdapterCallback callback;

    @Override
    public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
        if (callback != null) callback.onItemClick(taskItems[position].getTaskId());
    }


    interface TaskGridAdapterCallback {
        public void onItemClick(int taskId);
    }

    public void setCallback(TaskGridAdapterCallback callback) {
        this.callback = callback;
    }

    public TaskGridItemViewAdapter(@NonNull Context context, Task[] taskItems) {
        this.taskItems = taskItems;
    }

    @Override
    public int getCount() {
        return taskItems.length;
    }

    @Override
    public Object getItem(int i) {
        return taskItems[i];
    }

    @Override
    public long getItemId(int i) {
        return taskItems[i].getTaskId();
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {
        TaskGridItemView gridItem = (TaskGridItemView) view;
        if (gridItem == null) {
            gridItem = new TaskGridItemView(parent.getContext());
        }

        gridItem.setListItemBackground(
                ColorUtility.getColorResId(
                        parent.getContext(),
                        taskItems[position].getTaskColor()
                )
        );

        gridItem.setListItemIcon(
                IconUtility.getResIdFromEnum(
                        parent.getContext(),
                        taskItems[position].getTaskIcon()
                )
        );

        gridItem.setListItemText(taskItems[position].getTaskName());

        return gridItem;
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/tasklistview/TaskGridItemViewAdapter.kt
package com.wiseassblog.samsaradayplanner.ui.tasklistview

import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.wiseassblog.samsaradayplanner.common.getColorResId
import com.wiseassblog.samsaradayplanner.common.getResIdFromEnum
import com.wiseassblog.samsaradayplanner.domain.Task

internal class TaskGridItemViewAdapter(private val taskItems: Array<Task>) :
    BaseAdapter() {

    override fun getCount(): Int {
        return taskItems.size
    }

    override fun getItem(i: Int): Any {
        return taskItems[i]
    }

    override fun getItemId(i: Int): Long {
        return taskItems[i].taskId.toLong()
    }


    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        var gridItem: TaskGridItemView? = view as TaskGridItemView?

        if (gridItem == null) gridItem = TaskGridItemView(parent.context)

        gridItem = TaskGridItemView(parent.context)

        gridItem.setListItemBackground(
            getColorResId(
                parent.context,
                taskItems[position].taskColor
            )
        )
        gridItem.setListItemIcon(
            getResIdFromEnum(
                parent.context,
                taskItems[position].taskIcon
            )
        )
        gridItem.setListItemText(taskItems[position].taskName)
        return gridItem
    }
}